libs/datamodel/parser-database/src/walkers/model.rs

```diff
@@ -272,4 +273,43 @@ impl<'db> ModelWalker<'db> {
         self.inline_relations_from()
             .filter_map(|relation| relation.as_complete())
     }
+
+    /// How fields and arguments are indented in the model.
+    pub fn indentation(self) -> IndentationType {
+        let field = match self.scalar_fields().last() {
+            Some(field) => field,
+            None => return IndentationType::default(),
+        };
+
+        let src = self.db.source();
+        let start = field.ast_field().span.start;
+
+        let mut spaces = 0;
+
+        for i in (0..start).rev() {
+            match src.chars().nth(i) {
```

Comment:
this is quadratic, in the pathological case of a deeply indented field it could blow up